fetch_raw_data.py

```diff
@@ -0,0 +1,111 @@
+        pipeline_config = json.load(f)
+
+        rapid_pro_domain = pipeline_config["RapidProDomain"]
+        rapid_pro_token_file_url = pipeline_config["RapidProTokenFileURL"]
+        rapid_pro_test_contact_uuids = pipeline_config["RapidProTestContactUUIDs"]
+
+    # Fetch the Rapid Pro Token from the Google Cloud Storage URL
+    parsed_rapid_pro_token_file_url = urlparse(rapid_pro_token_file_url)
+    assert parsed_rapid_pro_token_file_url.scheme == "gs", "RapidProTokenFileURL needs to be a gs URL " \
+                                                           "(i.e. of the form gs://bucket-name/file)"
+    bucket_name = parsed_rapid_pro_token_file_url.netloc
+    blob_name = parsed_rapid_pro_token_file_url.path.lstrip("/")
+
+    print(f"Downloading Rapid Pro token from file '{blob_name}' in bucket '{bucket_name}'...")
+    storage_client = storage.Client.from_service_account_json(google_cloud_credentials_file_path)
+    credentials_bucket = storage_client.bucket(bucket_name)
+    credentials_blob = credentials_bucket.blob(blob_name)
+    rapid_pro_token = credentials_blob.download_as_string().strip().decode("utf-8")
+    print("Downloaded Rapid Pro token.")
+
+    with open(phone_number_uuid_table_path) as f:
+        phone_number_uuid_table = PhoneNumberUuidTable.load(f)
+
+    rapid_pro = RapidProClient(rapid_pro_domain, rapid_pro_token)
+    raw_contacts = rapid_pro.get_raw_contacts()
+
+    # Download all the runs for each of the radio shows
+    for show in SHOWS:
+        output_file_path = f"{raw_data_dir}/{show}.json"
+        print(f"Exporting show '{show}' to '{output_file_path}'...")
+
+        flow_id = rapid_pro.get_flow_id(show)
+        raw_runs = rapid_pro.get_raw_runs_for_flow_id(flow_id)
+        raw_contacts = rapid_pro.update_raw_contacts_with_latest_modified(raw_contacts)
+        traced_runs = rapid_pro.convert_runs_to_traced_data(
+            user, raw_runs, raw_contacts, phone_number_uuid_table, rapid_pro_test_contact_uuids)
+
+        with open(phone_number_uuid_table_path, "w") as f:
+            phone_number_uuid_table.dump(f)
+
+        IOUtils.ensure_dirs_exist_for_file(output_file_path)
+        with open(output_file_path, "w") as f:
+            TracedDataJsonIO.export_traced_data_iterable_to_json(traced_runs, f, pretty_print=True)
+
+    # Download all the runs for each of the surveys
+    for survey in SURVEYS:
+        output_file_path = f"{raw_data_dir}/{survey}.json"
+        print(f"Exporting survey '{survey}' to '{output_file_path}'...")
+
+        flow_id = rapid_pro.get_flow_id(survey)
+        raw_runs = rapid_pro.get_raw_runs_for_flow_id(flow_id)
+        raw_contacts = rapid_pro.update_raw_contacts_with_latest_modified(raw_contacts)
+        traced_runs = rapid_pro.convert_runs_to_traced_data(
+            user, raw_runs, raw_contacts, phone_number_uuid_table, rapid_pro_test_contact_uuids)
+        traced_runs = rapid_pro.coalesce_traced_runs_by_key(user, traced_runs, "avf_phone_id")
```

Comment:
(fyi when you switch to a released version of RapidProTools this function needs to move to the pipeline (see https://github.com/AfricasVoices/Project-ADSS/blob/7a582ab0cb7f0d2d03d5ec9819cfc79d81304fe5/generate_outputs.py#L114))